### PR TITLE
implemented creating work hours with tech create

### LINKED ISSUE #178: Register New Employee tab: Create New employee Account Along With a Schedule (Admin)
### **🏁 Setup**
- [ ] **Assign** yourself to this issue 
- [ ]  Read [Create New Employee Account Along With a Schedule](https://github.com/McGill-ECSE321-Winter2021/project-group-09/wiki/Detailed-Use-Cases#use-case-create-new-employee-account-along-with-a-schedule)

### **✅ To do:**
- [ ] **Draw** page mockup using [Whimsical](https://whimsical.com/repairshop-6v7GHoRVFfSSvyvUdeHaQ8) 
(After clicking on the link, you must sign in with your Google Account)

**Frontend**
- [ ] Enter the new employee's name, email, password, phone number and availability 
- [ ] Dropdown menu to select admin or technician
- [ ] Implement frontend

### **🆘 If you need help for the issue you’re working on:**

**1) Write** comments in this issue 
**2) Add** label image
**3) Copy** the issue link and **paste** it to Discord's `#🐛issues` channel
**4)** When solved, **strike** the message in `#🐛issues`, **mark it** as `(SOLVED)`  + **remove** label `help wanted` + **write** comments in this issue 

### *Whenever something is done write a comment below.*

### PR DESCRIPTION
**Pull request by JDziewonski98**
Looks ugly but it works, was surprisingly big pain, can make it prettier but approve the functionality plz
Only creates timeslots where you validly entered both start and end time
image

### CODE REVIEW

**Review by Shaswata79 — APPROVED** [review]

**Comment by T-Syme** [comment]
I tried testing it and here are my suggestions from that:

The functionality (actually adding the schedule) works well!

When adding the work hours, I see that they are being saved as UTC time regardless of the time zone, which could be confusing if everything else is displayed in local time.
image

As well, I think the system can handle times that don't end every half hour (I pretty sure it shouldn't affect booking an appointment), but it might be nice to restrict it to times like 09:00, 17:30 instead of allowing times like 10:22, although I don't know if that's easy to do with the current setup. 

I had to do something similar to this to add individual timeslots, so if you want to reuse some functions, check out [#192](https://github.com/McGill-ECSE321-Winter2021/project-group-09/pull/192) 
I converted the dates to the time in milliseconds and it is automatically interpreted correctly by the backend. I also used regular textboxes with regex for the times

File: RepairShop-Frontend/src/components/Register.vue [hunk 1]
```diff
@@ -128,12 +221,70 @@ export default {
...
+    //formats the time to timestamp with the appropriate date based on 'day' param ( 0 : sunday ... 6: saturday)
+    formatTimeToTimestamp(time, day) {
+      let today = new Date();
+      let todaysDayOfWeek = today.getDay();
+      today.setDate(today.getDate() - (todaysDayOfWeek - day));
+      let dd = String(today.getDate()).padStart(2, "0");
+      let mm = String(today.getMonth() + 1).padStart(2, "0");
+      let yyyy = today.getFullYear();
+      today = yyyy + "-" + mm + "-" + dd;
+      return today + "T" + time + "+00:00";
```
Comment (T-Syme) [inline RepairShop-Frontend/src/components/Register.vue:254]: We may want to change this. See other comment

File: RepairShop-Frontend/src/components/Register.vue [hunk 2]
```diff
@@ -150,6 +301,7 @@ export default {
           console.error("Invalid usertype");
           return;
       }
+      console.log(this.$root.$data.token);
```
Comment (T-Syme) [inline RepairShop-Frontend/src/components/Register.vue]: Not a big deal, but we may want to remove the console.logs later

**Review by T-Syme — COMMENTED** [review]
See the main comment above

Functionality looks good!

**Review by sandyl289 — COMMENTED** [review]
- Registering a customer works! 💯 👍🏼 
- I don't know if it's only me, but I am unable to register a second technician or an admin right after registering one (see screenshots below (TEST 2 & TEST 3). Does anyone else have the same problem as me?
- I get the message `email taken`

# Testing

## TEST 1: Register 1 technician ✔️

### 1) Register New User page (After signing in as Admin) ✔️

image
image

### 2) After `Submit` (with successful message) ✔️
image
image


### Not really important: but I think it would be nice if we have all the text fields cleared up right after we submit 
image

### POSTMAN (verify if all workhours have been added)
image

## TEST 2: Register 2nd technician

### 1) Register 
image
image

### 2) `Submit` ⚠️
⚠️For some reason, I get `email taken` if I try to register another technician right after.
I tried to change the email to another one and I still get the same message.
image

### 3) If try to log in as `someoneElse@gmail.com`
image


## TEST 3: Register New Admin

### 1) Fill out information
image

### 2) `Submit` button⚠️
⚠️ I get the same message as before. It seems like we can't register another user right after. 
image

## TEST 4: Register Customer ✔️

### 1) Register + submit✔️
image

### 2) Log In as the new `helloCustomer@gmail.com` customer ✔️
image

**Comment by JDziewonski98** [comment]
> * Registering a customer works! 💯 👍🏼
> * I don't know if it's only me, but I am unable to register a second technician or an admin right after registering one (see screenshots below (TEST 2 & TEST 3). Does anyone else have the same problem as me?
> * I get the message `email taken`
> 
> # Testing
> ## TEST 1: Register 1 technician ✔️
> ### 1) Register New User page (After signing in as Admin) ✔️
> image
> image
> 
> ### 2) After `Submit` (with successful message) ✔️
> image
> image
> 
> ### Not really important: but I think it would be nice if we have all the text fields cleared up right after we submit
> image
> 
> ### POSTMAN (verify if all workhours have been added)
> image
> 
> ## TEST 2: Register 2nd technician
> ### 1) Register
> image
> image
> 
> ### 2) `Submit` ⚠️
> ⚠️For some reason, I get `email taken` if I try to register another technician right after.
> I tried to change the email to another one and I still get the same message.
> image
> 
> ### 3) If try to log in as `someoneElse@gmail.com`
> image
> 
> ## TEST 3: Register New Admin
> ### 1) Fill out information
> image
> 
> ### 2) `Submit` button⚠️
> ⚠️ I get the same message as before. It seems like we can't register another user right after.
> image
> 
> ## TEST 4: Register Customer ✔️
> ### 1) Register + submit✔️
> image
> 
> ### 2) Log In as the new `helloCustomer@gmail.com` customer ✔️
> image

Thanks for the detailed tests. I tried creating multiple accounts in succession and it worked, couldn't replicate your bug

**Pull request merged**